code/lib/core-server/src/build-dev.ts

```diff
@@ -102,6 +105,16 @@ export async function buildDevStandalone(
     getManagerBuilder(),
   ]);
 
+  if (builderName.includes('builder-vite')) {
+    const mainJsPath = serverResolve(resolve(options.configDir || '.storybook', 'main')) as string;
+    if (/\.c[jt]s$/.test(mainJsPath)) {
+      deprecate(
+        dedent(`Using the .cjs or .cts extension for your main config file is deprecated with Vite.
```

Comment:
It's a bit more complicated than this so maybe just link to the `MIGRATION.md`

Reply:
What you do think about _"Using CommonJS in your main config file is deprecated with Vite."_?

Reply:
that sounds good to me

Reply:
```suggestion
        dedent(`Using CommonJS in your main configuration file is deprecated with Vite."
```